editor.planx.uk/src/pages/Preview/ReconciliationPage.tsx

```diff
@@ -35,16 +37,37 @@ const ReconciliationPage: React.FC<Props> = ({
   buttonText,
   onButtonClick,
 }) => {
-  const [flow, changeAnswer] = useStore((state) => [
-    state.flow,
-    state.changeAnswer,
-  ]);
+  const [flow, hasSections, sectionNodes, sectionStatuses, changeAnswer] =
+    useStore((state) => [
+      state.flow,
+      state.hasSections,
+      state.sectionNodes,
+      state.sectionStatuses,
+      state.changeAnswer,
+    ]);
 
   const sortedBreadcrumbs = sortBreadcrumbs(
     data?.reconciledSessionData?.breadcrumbs,
     flow
   );
 
+  // Assumes that reconciliation API has handled removing the associated section breadcrumb for each updated node
+  //   and returned them as part of `removedBreadcrumbs` so we can accurately set sectionStatuses
+  const updatedSectionNodeIds: string[] = [];
+  if (data?.removedBreadcrumbs) {
+    Object.keys(data.removedBreadcrumbs)?.forEach((nodeId: string) => {
+      if (flow[nodeId]?.type === TYPES.Section) {
+        updatedSectionNodeIds.push(nodeId);
+      }
+    });
+  }
```

Comment:
nit/suggestion: as mentioned above, you could pass in an array of all affected node ids instead of filtering for sectionIds specifically (finding section ids can be `sectionStatuses` job). But filtering to only the nodeIds you care about makes sense too.

```suggestion
const affectedNodeIds =  data.removedBreadcrumbs.map((crumb) => crumb.id)
```

Reply:
addressed this is https://github.com/theopensystemslab/planx-new/pull/1535/commits/76c388f2bab1fa8efca4281bc4fa87037e49e937 !